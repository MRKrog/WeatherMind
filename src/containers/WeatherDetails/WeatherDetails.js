import React, { Component } from 'react';
import { connect } from 'react-redux';
import { NavLink } from 'react-router-dom';
export class WeatherDetails extends Component {

  render(){
    return (
      <div className="WeatherDetails">
        <div className="WeatherDetails-Content">
          <NavLink className="Details-Close" to="/">
            <i className="fas fa-window-close"></i>
          </NavLink>
          <div className="Details-Content">
            <div className="Details-Row">
              <section className="Details-Col">
                <label>Today's Summary</label>
                <h5 className="Summary">{this.props.today.summary}</h5>
              </section>
            </div>
            <div className="Details-Row">
              <section className="Details-Col">
                <label>Sunrise</label>
                <h5 className="rise">{this.props.today.sunriseTime}</h5>
              </section>
              <section className="Details-Col">
                <label>Sunset</label>
                <h5 className="rise">{this.props.today.sunsetTime}</h5>
              </section>
            </div>
            <div className="Details-Row">
              <section className="Details-Col">
                <label>Wind speed</label>
                <h5 className="wind">{this.props.today.windSpeed} mph</h5>
              </section>
              <section className="Details-Col">
                <label>humidity</label>
                <h5 className="humidity">{this.props.today.humidity}%</h5>
              </section>
            </div>
            <div className="Details-Row">
              <section className="Details-Col">
                <label>visibility</label>
                <h5 className="visibility">{this.props.today.visibility} mi</h5>
              </section>
              <section className="Details-Col">
                <label>Uv index</label>
                <h5 className="uvIndex">{this.props.today.uvIndex}</h5>
              </section>
            </div>
            <div className="Details-Row">
              <section className="Details-Col">
                <label>Chance of rain</label>
                <h5 className="precipProbability">{this.props.today.precipProbability}%</h5>
              </section>
              <section className="Details-Col">
                <label>Moon phase</label>
                <h5 className="moonPhase">{this.props.today.moonPhase}%</h5>
              </section>
            </div>
          </div>
        </div>
      </div>
    )
  }
}

export const mapStateToProps = (state) => ({
  current: state.current,
  today: state.today,
})


export default connect(mapStateToProps)(WeatherDetails)

// apparentTemperature:
// 66
// humidity:
// 0.18
// icon:
// "partly-cloudy-day"
// pressure:
// 1003.69
// summary:
// "Partly Cloudy"
// temperature:
// 66
// time:
// 1555276440
// uvIndex:
// 4
// visibility:
// 3.82
