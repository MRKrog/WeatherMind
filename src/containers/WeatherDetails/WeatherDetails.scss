.WeatherDetails {
  background: #231f20d4;
  position: absolute;
  width: 100%;
  height: 100%;
  top: 0px;
  color: #fff;

  .WeatherDetails-Content {
    position: fixed;
    top: 50%;
    left: 50%;
    width: 300px;
    height: auto;
    background: #1d162b;
    transform: translate(-50%,-50%);
    box-shadow: 0px 2px 10px 3px #0000006b;
    display: flex;
    flex-direction: column;
    padding: 1em;
    width: 340px;

    .Details-Close {
      position: absolute;
      top: 0px;
      right: 0px;
      color: #fff;
    }

    .Details-Content {
      display: flex;
      flex-direction: column;

      .Details-Row {
        display: flex;
        flex-direction: row;
      }
    }
  }
}
