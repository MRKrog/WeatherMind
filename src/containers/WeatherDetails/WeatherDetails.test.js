import React, { Component } from 'react';
import { WeatherDetails, mapStateToProps } from './WeatherDetails';
import { shallow } from 'enzyme';

const mockTodayData = {
  humidity: 39,
  moonPhase: 33,
  precipProbability: 13,
  summary: "Partly cloudy throughout the day.",
  sunriseTime: "6:24 AM",
  sunsetTime: "7:37 PM",
  tempHigh: 67,
  tempLow: 40,
  time: "Sunday",
  uvIndex: 6,
  visibility: 4.32,
  windSpeed: 1.24,
}

describe('WeatherDetails', () => {

  describe('WeatherDetails Component', () => {
    let wrapper;

    beforeEach(() => {
      wrapper = shallow(
        <WeatherDetails today={mockTodayData}/>
      )
    })

    it('should match the snapshot', () => {
      expect(wrapper).toMatchSnapshot()
    })
  })

  describe('mapStateToProps', () => {
    it('should ', () => {

    })
  })

})
