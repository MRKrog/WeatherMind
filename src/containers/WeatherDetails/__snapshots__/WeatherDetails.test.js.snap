// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WeatherDetails WeatherDetails Component should match the snapshot 1`] = `
<div
  className="WeatherDetails"
>
  <div
    className="WeatherDetails-Content"
  >
    <NavLink
      className="Details-Close"
      to="/"
    >
      <i
        className="fas fa-window-close"
      />
    </NavLink>
    <div
      className="Details-Content"
    >
      <div
        className="Details-Row"
      >
        <section
          className="Details-Col"
        >
          <label>
            Today's Summary
          </label>
          <h5
            className="Summary"
          >
            Partly cloudy throughout the day.
          </h5>
        </section>
      </div>
      <div
        className="Details-Row"
      >
        <section
          className="Details-Col"
        >
          <label>
            Sunrise
          </label>
          <h5
            className="rise"
          >
            6:24 AM
          </h5>
        </section>
        <section
          className="Details-Col"
        >
          <label>
            Sunset
          </label>
          <h5
            className="rise"
          >
            7:37 PM
          </h5>
        </section>
      </div>
      <div
        className="Details-Row"
      >
        <section
          className="Details-Col"
        >
          <label>
            Wind speed
          </label>
          <h5
            className="wind"
          >
            1.24
             mph
          </h5>
        </section>
        <section
          className="Details-Col"
        >
          <label>
            humidity
          </label>
          <h5
            className="humidity"
          >
            39
            %
          </h5>
        </section>
      </div>
      <div
        className="Details-Row"
      >
        <section
          className="Details-Col"
        >
          <label>
            visibility
          </label>
          <h5
            className="visibility"
          >
            4.32
             mi
          </h5>
        </section>
        <section
          className="Details-Col"
        >
          <label>
            Uv index
          </label>
          <h5
            className="uvIndex"
          >
            6
          </h5>
        </section>
      </div>
      <div
        className="Details-Row"
      >
        <section
          className="Details-Col"
        >
          <label>
            Chance of rain
          </label>
          <h5
            className="precipProbability"
          >
            13
            %
          </h5>
        </section>
        <section
          className="Details-Col"
        >
          <label>
            Moon phase
          </label>
          <h5
            className="moonPhase"
          >
            33
            %
          </h5>
        </section>
      </div>
    </div>
  </div>
</div>
`;
